lib/extensions/formattable/coded_value.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module ConnectProto
+  module Extensions
+    module Formattable
+      module CodedValue
+        def to_hl7
+          [
+            value.presence,
+            description.presence,
+            code_set.presence
+          ].compact.join('^').squeeze(' ').strip
```

Comment:
I don't think you want to compact... if one of the values is blank, you still need it to keep it's place, don't you? E.g.
```ruby
{
  value: 'abc',
  code_set: 'LOINC'
}
```
should convert to
```
abc^^LOINC
```
and not
```
abc^LOINC
```
Shouldn't it?

Reply:
You're right, we should keep a blank value